src/utils/DepositUtils.ts

```diff
@@ -0,0 +1,73 @@
+import { assert } from "chai";
+import { SpokePoolClient } from "../clients";
+import { Fill, DepositWithBlock, Deposit, CachingMechanismInterface } from "../interfaces";
+import { validateFillForDeposit } from "./FlowUtils";
+import { getCurrentTime } from "./TimeUtils";
+import { isDefined } from "./TypeGuards";
+import { getDepositInCache, getDepositKey, setDepositInCache } from "./CachingUtils";
+import { DEFAULT_CACHING_TTL } from "../constants";
+
+// Load a deposit for a fill if the fill's deposit ID is outside this client's search range.
+// This can be used by the Dataworker to determine whether to give a relayer a refund for a fill
+// of a deposit older or younger than its fixed lookback.
+
+/**
+ * Attempts to resolve a deposit for a fill. If the fill's deposit Id is within the spoke pool client's search range,
+ * the deposit is returned immediately. Otherwise, the deposit is queried first from the provided cache, and if it is
+ * not found in the cache, it is queried from the spoke pool client. If the deposit is found, it is cached before
+ * being returned.
+ * @param spokePoolClient The spoke pool client to use to query the deposit.
+ * @param fill The fill to resolve a deposit for.
+ * @param cache An optional cache to use to store the deposit. Optional.
+ * @returns The deposit for the fill, or undefined if the deposit could not be found.
+ * @throws If the fill's origin chain ID does not match the spoke pool client's chain ID.
+ * @throws If the spoke pool client has not been updated.
+ */
+export async function queryHistoricalDepositForFill(
```

Comment:
This is copied from your latest commit in `relayer-v2` right?

Reply:
This is coped from relayer-v2 on prod with the change that we are now using the caching mechanism interface instead of specifically the `RedisClient`